rust-src/id/src/utils.rs

```diff
@@ -26,15 +26,14 @@ pub fn commitment_to_share<C: Curve>(
     share_number: &C::Scalar,
     coeff_commitments: &[Commitment<C>],
 ) -> Commitment<C> {
-    // TODO: This would benefit from multiexponentiation.
-    let mut cmm_share_point: C = C::zero_point();
-    // Horner's scheme in the exponent
-    for cmm in coeff_commitments.iter().rev() {
-        // FIXME: This would benefit from multiexponentiation.
-        cmm_share_point = cmm_share_point.mul_by_scalar(&share_number);
-        cmm_share_point = cmm_share_point.plus_point(cmm);
+    let n = coeff_commitments.len();
```

Comment:
I would recommend checking if n == 0 and returning early since `multiexp` will do needless work in that case.